src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -34,52 +33,44 @@ namespace System.IO.Enumeration
             entry._fullPath = ReadOnlySpan<char>.Empty;
             entry._fileName = ReadOnlySpan<char>.Empty;
 
-            // IMPORTANT: Attribute logic must match the logic in FileStatus
+            entry._status = default;
+            entry._status.InvalidateCaches();
+
+            bool isDirectory = directoryEntry.InodeType == Interop.Sys.NodeType.DT_DIR;
+            bool isSymlink = directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK;
```

Comment:
I think I need to simplify all the InodeType checks in this method:

```suggestion
            bool isSymlink = directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK;
            bool isUnknown = directoryEntry.InodeType == Interop.Sys.NodeType.DT_UNKNOWN;
```